samples/csharp/MicrosoftEduImportFromGoogle/Import.cs

```diff
@@ -0,0 +1,323 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Graph.Beta;
+using Microsoft.Graph.Beta.Models;
+using MicrosoftEduImportFromGoogle.Models;
+using MicrosoftGraphSDK;
+
+namespace MicrosoftEduImportFromGoogle
+{
+    /// <summary>
+    /// Microsoft endpoints needed to import assignments, classwork modules and resources
+    /// </summary>
+    internal class Import
+    {
+        private readonly IConfiguration _config;
+        public GraphServiceClient graphServiceClient;
+        public Import(IConfiguration configuration)
+        {
+            this._config = configuration;
+        }
+
+        /// <summary>
+        /// Authorizes the application and creates a Microsoft Graph client
+        /// </summary>
+        /// <returns></returns>
+        public void AuthorizeApp()
+        {
+            // App-only scenario
+            this.graphServiceClient = GraphClient.GetApplicationClient(_config["microsoftTenantId"], _config["microsoftClientId"], _config["microsoftSecret"]);
+
+            // Use the line below for App+user scenario
+            // this.graphServiceClient = GraphClient.GetDelegateClient(_config["microsoftClientId"]);
+        }
+
+        /// <summary>
+        /// Returns a list of Teams that the requesting user is permitted to view
+        /// </summary>
+        /// <param name="userId">User id</param>
+        /// <returns>List<Microsoft.Graph.Beta.Models.Team></returns>
+        public List<Microsoft.Graph.Beta.Models.Team> GetUserClasses(string userId)
+        {
+            Console.WriteLine("* Fetching classes from Microsoft Teams...");
+            return graphServiceClient
+                    .Users[userId]
+                    .JoinedTeams
+                    .GetAsync()
+                    .Result.Value.ToList();
+        }
+
+        /// <summary>
+        /// Returns a list of classes that the requesting application is permitted to view
+        /// </summary>
+        /// <returns>List<EducationClass></returns>
+        public List<EducationClass> GetClasses()
+        {
+            Console.WriteLine("* Fetching classes from Microsoft Teams...");
+            return graphServiceClient.Education.Classes
+                    .GetAsync()
+                    .Result.Value.ToList();
+        }
+
+        /// <summary>
+        /// Maps Google entities to create assignments
+        /// </summary>
+        /// <param name="courseWorks">Array of courseWorks to import</param>
+        /// <param name="classId">User class id</param>
+        /// <param name="exporterInstance">Instance of the Export class</param>
+        /// <returns>List<string></returns>
+        public async Task<List<string>> MapAndCreateAssignments(CourseWork[] courseWorks, string classId, Export exporterInstance)
```

Comment:
I am not understanding the mapping here in this method and the next...is the below correct?

 - MapAndCreateAssignments maps 1 Google CourseWork entity to 1 Microsoft Assignment entity
 - MapAndCreateModules maps 1 Google CourseWork entity to 1 Microsoft ClassworkModule entity

If yes, why are we mapping the same entity type in Google to different Microsoft entities?

Reply:
Correct mapping is:

- MapAndCreateAssignments maps 1 Google **CourseWork** entity to 1 Microsoft **Assignment** entity
- MapAndCreateModules maps 1 Google **CourseWorkMaterial** entity to 1 Microsoft **ClassworkModule** entity

Reply:
Ok thanks. There is another Google entity type Topic. I was expecting Google Topic to map to Microsoft ClassworkModule

Reply:
yes, but the topic entity is like a category for the assignment